src/python/pants/backend/python/packaging/pyoxidizer/rules.py

```diff
@@ -51,91 +52,80 @@ class PyOxidizerFieldSet(PackageFieldSet):
 async def package_pyoxidizer_binary(
     pyoxidizer: PyOxidizer, field_set: PyOxidizerFieldSet
 ) -> BuiltPackage:
-    logger.debug(f"Incoming package_pyoxidizer_binary field set: {field_set}")
-    targets = await Get(Targets, DependenciesRequest(field_set.dependencies))
-    target = targets[0]
-
-    logger.debug(f"Received these targets inside pyox targets: {target.address.target_name}")
-
-    packages = await Get(
-        FieldSetsPerTarget,
-        FieldSetsPerTargetRequest(PackageFieldSet, [target]),
+    direct_deps, pyoxidizer_pex = await MultiGet(
+        Get(Targets, DependenciesRequest(field_set.dependencies)),
+        Get(
+            Pex,
+            PexRequest(
+                output_filename="pyoxidizer.pex",
+                internal_only=True,
+                requirements=pyoxidizer.pex_requirements(),
+                interpreter_constraints=pyoxidizer.interpreter_constraints,
+                main=pyoxidizer.main,
+            ),
+        ),
     )
-    logger.debug(f"Retrieved the following FieldSetsPerTarget {packages}")
 
+    deps_field_sets = await Get(
+        FieldSetsPerTarget, FieldSetsPerTargetRequest(PackageFieldSet, [direct_deps[0]])
+    )
     built_packages = await MultiGet(
-        Get(BuiltPackage, PackageFieldSet, field_set) for field_set in packages.field_sets
+        Get(BuiltPackage, PackageFieldSet, field_set) for field_set in deps_field_sets.field_sets
     )
```

Comment:
Given that this is going to be flattened into a list of wheels, it seems like it could include all of the `direct_deps`, rather than only the first? That would maybe make it more challenging to inherit the entrypoint from the distribution later though, so maybe best to error if there is more than one dep.

Reply:
Indeed. In fact I made that change, but then I reverted it to save for a dedicated followup where I can reason more directly about this all.